source/test/TestApp/PlayFabGroupsTest.cpp

```diff
@@ -0,0 +1,85 @@
+
+#include "TestAppPch.h"
+
+#include "PlayFabGroupsTest.h"
+#include "playfab/PlayFabApiSettings.h"
+#include "playfab/PlayFabSettings.h"
+
+using namespace PlayFab;
+
+namespace PlayFabUnit
+{
+
+    // GROUPS API
+    void PlayFabGroupsTest::GroupsApiTest(TestContext& testContext)
+    {
+        auto settings = std::make_shared<PlayFabApiSettings>();
+
+        auto req = ClientModels::LoginWithCustomIDRequest();
+        req.CustomId = PlayFabSettings::buildIdentifier;
+
+        clientApi->LoginWithCustomID(req, Callback(&PlayFabGroupsTest::GroupsTestLoginCallback), Callback(&PlayFabGroupsTest::GroupsTestLoginFailedCallback), &testContext);
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginCallback(const ClientModels::LoginResult& result, void* customData)
+    {
+        auto groupApi = std::make_shared<PlayFabGroupsInstanceAPI>(result.authenticationContext);
+        auto req = GroupsModels::CreateGroupRequest();
+        req.CustomTags = std::map<std::string, std::string>();
+
+        // TODO Bug 29786037: this map is required to be filled to not get a 500 error with the CreateGroup api call
+        req.CustomTags.insert(std::pair<std::string, std::string>("One", "Two"));
+
+        req.GroupName = GenerateUuidV4();
+        groupApi->CreateGroup(req, Callback(&PlayFabGroupsTest::GroupsTestGroupCallback), Callback(&PlayFabGroupsTest::GroupsTestLoginFailedCallback), customData);
+    }
+
+    void PlayFabGroupsTest::GroupsTestGroupCallback(const GroupsModels::CreateGroupResponse&, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Pass();
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginFailedCallback(const PlayFabError& error, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Fail("Expected Group Login to succeed. Got error " + error.ErrorMessage);
+    }
+
+    void PlayFabGroupsTest::AddTests()
+    {
+        AddTest("CreateGroupApi", &PlayFabGroupsTest::GroupsApiTest);
```

Comment:
I think a better plan here would be to separate the login step into it's own test, and put two tests here.
That gives each call an independent test-timeout, and is a little safer in terms of validating each call independently.
It's true that the "login" test isn't a test of groups, but as we add tests, we don't want to duplicate login logic.

For individual suites we've been very comfortable with the in-suite ordering being strict and reliable.

Reply:
Other login tests in previous test suites have been on a per test basis. 

If we want to combine logins for additional tests in the future we can do it then.

Reply:
Actually, That is my point. Every other suite has login as a separate test & step. This is the only suite that would combine them, and I don't think it's a good idea. And, since this is a new suite, I don't think we should create it from scratch the wrong way, regardless.